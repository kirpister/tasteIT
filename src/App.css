@import url('https://fonts.googleapis.com/css2?family=Coda&family=Sriracha&family=Ubuntu&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  overscroll-behavior: none;
}

:root {
  --main-font:'Ubuntu', sans-serif;
  --shrimp-color: rgb(243, 156, 154);
  --white-color: rgba(255, 255, 255, 0.7);
  --logo-font: 'Coda', cursive;
}

/* body {
  background-color: rgb(15, 17, 20);
} */

a {
  text-decoration: none;
}

button {
  width: 130px;
  padding: 10px;
  border-radius: 5px;
  background-color: transparent;
  border: 1px solid var(--shrimp-color);
  color: var(--shrimp-color);
  cursor: pointer;
  font-family: var(--main-font);
}

main {
  min-height: calc(100vh - 110px);
}

header {
  height: 70px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

header, footer {
  background-image: url(./assets/bg_marble.jpg);
  background-color: rgba(15, 17, 20, 0.2);
  background-blend-mode: multiply;
  background-position: bottom;
}

footer {
  height: 40px;
  text-align: center;
  color: var(--white-color);
  padding: 10px;
}

h1 {
  font-size: 3.5rem;
  color: var(--white-color);
  letter-spacing: 0.5rem;
  font-family: var(--logo-font);
}

h2 {
  margin-left: 2.5rem;
  font-size: 2rem;
  letter-spacing: 0.5rem;
  display: flex;
  justify-content: flex-start;
  font-family: var(--logo-font);
  color: var(--white-color);
}

/* h3 {
  font-family: var(--main-font);
  font-size: 1.8rem;
  letter-spacing: 2px;
  color: rgba(255, 255, 255, 0.7);
  margin: 2rem 0 6rem 0;
} */

nav ul li {
  display: inline-block;
}

nav ul li a {
  list-style-type: none;
  color: var(--white-color);
  text-decoration: none;
  padding: 1rem;
  letter-spacing: 2px;
  margin-right: 2rem;
}

nav a:active {
  color: var(--shrimp-color);
}

/* HERO BANNER */

/* .video {
  width: 100vw;
  height: auto;
  z-index: -1;
  object-fit: cover;
  position: relative;
}

.intro-container {
  width: 40vw;
  height: 35vh;
  position: absolute;
  top: 22%;
  left: 10%;
}


/* HOME CARDS */

/* .main-container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 85%;
  height: 40vh;
  margin: 0 auto;
}

.home-card {
  width: 25%;
  height: 25vh;
  border-radius: 3px;
}

.home-card h4, p {
  color: rgba(255, 255, 255, 0.7);
  margin: 1.5rem 0 1.5rem 1.5rem;
}

.home-card small {
  color: var(--shrimp-color);
  margin: 3.5rem 0 1rem 1.5rem;
} */

/* ADD RECIPE */

/* .wrapper {
  height: 100vh;
  background-image: url(./assets/salmon.jpg);
  background-color: rgba(25, 24, 24, 0.706);
  background-attachment: fixed;
  
  background-position: top;
  background-size: cover;
  background-blend-mode: multiply;
}

.form-cont {
  margin: 5rem;
}

.form-cont button {
  margin: 0.7rem 0 .5rem 1.2rem;
}

.form-cont h3 {
  margin: 0 0 2rem 1rem;
} */

/* form {
  width: 30vw;
  height: 70vh;
}

input,
select {
  display: block;
  width: 320px;
  padding: 0.5rem;
  border-radius: 0.5rem;
  border: none;
  margin: 0 0 .5rem 1rem;
  background-color: rgba(105, 105, 105, 0.445);
  color: white;
  font-family: var(--main-font);
} */

/* label {
  font-size: 0.8rem;
  line-height: 1.2;
  color: rgba(255, 255, 255, 0.7);
  margin: 1rem 0 0.5rem 1.1rem;
  display: block;
}

textarea {
  width: 320px;
  height: 70px;
  margin: 0 0 0 1rem;
  border-radius: 0.5rem;
  background-color: rgba(105, 105, 105, 0.445);
  color: white;
} */
/* 
input,textarea,select:focus {
  border-color: transparent;
  outline: none;
}

.row {
  display: flex;
}

.input2 input {
  float: right;
  width: 150px;
}

.input1 input {
  width: 150px;
} */