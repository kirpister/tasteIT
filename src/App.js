import React, { useEffect } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import useLocalStorage from 'use-local-storage';

import './App.css';
import AddRecipe from './components/AddRecipe';
import BrowseRecipes from './components/BrowseRecipes';
import Recipe from './components/Recipe';
import Home from './components/Home';
import Layout from './components/Layout';
import Header from './components/Header';
import ToTop from './components/ToTop';

const App = () => {

  const defaultDark = window.matchMedia('(prefers-color-scheme: dark)')
  const [theme, setTheme] = useLocalStorage('theme', defaultDark ? 'dark' : 'light');

  const switchTheme = () => {
    const newTheme = theme === 'light' ? 'dark' : 'light';
    setTheme(newTheme);
  }

  useEffect(() => {
    document.documentElement.setAttribute("data-theme", theme);
  }, [theme]);

  return (

    
    <BrowserRouter>
      <ToTop />
      <div className='App'>
        <Header switchTheme={switchTheme}/>
        <Routes>
         
          <Route path='/' element={<Layout />}>
            <Route index element={<Home />} />
            <Route path='browserecipes' element={<BrowseRecipes />} />
            <Route path='browserecipes/:singlerecipe' element={<Recipe />}  />
            <Route path='add' element={<AddRecipe />} />
          </Route>
        </Routes>

      </div>
      
      </BrowserRouter>
      
  );
}

export default App;
