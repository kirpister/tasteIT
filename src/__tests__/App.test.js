import React from "react";
import ReactDOM from "react-dom";
import '../matchMedia';
import App from '../App';

import renderer from 'react-test-renderer';


test("renders the correct content", () => {

const root = document.createElement("div");
ReactDOM.render(<App />, root);

// testing the h1 heading
expect(root.querySelector("h1").textContent).toBe("tasteIT");

// testing the button text on home page
expect(root.querySelector("button").textContent).toBe("Browse Recipes");

});

describe('Jest snapshot testing suite', () => {
  test('Matches DOM Snapshot for App component', () => {
      const domTree = renderer.create(<App />).toJSON();

      expect(domTree).toMatchSnapshot();
  });
});












