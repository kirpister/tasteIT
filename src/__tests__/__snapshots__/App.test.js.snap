// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest snapshot testing suite Matches DOM Snapshot for App component 1`] = `
<div
  className="App"
>
  <header>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <h2>
        taste
        <b>
          IT
        </b>
      </h2>
    </a>
    <div
      className="mode-toggle"
    >
      <input
        className="checkbox"
        id="checkbox"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="checkbox-label"
        htmlFor="checkbox"
      >
        <img
          alt="sun"
          className="sun"
          src="sun.png"
        />
        <img
          alt="moon"
          className="moon"
          src="moon.png"
        />
        <span
          className="ball"
        />
      </label>
    </div>
    <nav>
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className=""
            href="/browserecipes"
            onClick={[Function]}
          >
            Recipes
          </a>
        </li>
        <li>
          <a
            className=""
            href="/add"
            onClick={[Function]}
          >
            Add new recipe
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main>
    <main>
      <video
        autoPlay={true}
        className="video"
        loop={true}
        muted={true}
      >
        <source
          src="video.mp4"
          type="video/mp4"
        />
      </video>
      <div
        className="intro-container"
      >
        <h1>
          tasteIT
        </h1>
        <h3>
          Travel the world from your kitchen!
        </h3>
        <a
          href="/browserecipes"
          onClick={[Function]}
        >
          <button>
            Browse Recipes
          </button>
        </a>
      </div>
      <div
        className="main-container"
      >
        <div
          className="home-card"
        >
          <h4>
            Browse Recipes
          </h4>
          <p>
            Need some inspiration? Browse here for recipes from around the world.
          </p>
          <a
            href="/browserecipes"
            onClick={[Function]}
          >
            <small>
              All Recipes
            </small>
          </a>
        </div>
        <div
          className="home-card"
        >
          <h4>
            Add Recipes
          </h4>
          <p>
            Wanna add your local favorite into our collection?
          </p>
          <a
            href="/add"
            onClick={[Function]}
          >
            <small>
              Add Recipes
            </small>
          </a>
        </div>
        <div
          className="home-card"
        >
          <h4>
            More about our projects
          </h4>
          <p>
            If you're interested in knowing more about our program visit BCH homepage.
          </p>
          <a
            href="http://bc.fi"
            rel="noreferrer"
            target="_blank"
          >
            <small>
              Business College Helsinki
            </small>
          </a>
        </div>
      </div>
    </main>
  </main>
  <footer>
    <small>
      Â© Made by Jenni at BCH Full Stack Developer program
    </small>
  </footer>
</div>
`;
